cmd/gitops/add/management-cluster/flag.go

```diff
@@ -0,0 +1,53 @@
+package mcluster
+
+import (
+	"time"
+
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/cobra"
+)
+
+const (
+	flagDryRun          = "dry-run"
+	flagInterval        = "refresh-interval"
+	flagName            = "name"
+	flagLocalPath       = "local-path"
+	flagRepositoryKind  = "repository-kind"
+	flagRepositoryeName = "repository-name"
+	flagServiceAccount  = "service-account"
+	flagTimeout         = "refresh-timeout"
+)
+
+type flag struct {
+	DryRun         bool
+	Interval       time.Duration
+	Name           string
+	LocalPath      string
+	RepositoryKind string
+	RepositoryName string
+	ServiceAccount string
+	Timeout        time.Duration
+}
+
+func (f *flag) Init(cmd *cobra.Command) {
+	cmd.Flags().BoolVar(&f.DryRun, flagDryRun, false, "Print files and directories instead of creating them")
+	cmd.Flags().DurationVar(&f.Interval, flagInterval, time.Minute, "Source synchronization interval")
+	cmd.Flags().StringVar(&f.Name, flagName, "", "Codename of the Management Cluster")
+	cmd.Flags().StringVar(&f.LocalPath, flagLocalPath, ".", "Path to the cloned GitOps repository")
+	cmd.Flags().StringVar(&f.RepositoryKind, flagRepositoryKind, "GitRepository", "Type of the GitOps repository")
```

Comment:
Should we even make this configurable? Considering we don't advise to change that in `gitops-template`

Reply:
I'm not sure to be honest, hardcoding it may be safer option, I didn't have any particular reason to make this configurable. Well, my reasoning was, `kgs` is statless and is unaware of any "source", so it just creates a bunch of files and directories under pointed location, but at the end it could be "Bucket" where user decides to store these files, so why not give him a chance to specify it 😄